### ISSUE TITLE
Sprites moving attached to nodes.

### ISSUE THREAD

**Issue by PlayMa256** [graphstream/gs-core#199]
I just wanna check if this is the expected behaviour for sprites. I've tested with a few scenarios, sprites under the nodes (with different z-indexes), sprites over nodes, etc..and all had the same result, as you can see on the example video below.
[Video Link](https://www.youtube.com/watch?v=QQDNWuWHirU)

I just want to make sure if its the correct behaviour or not, or maybe it can be a bug, i dont know. Is this correct? While clicking on the sprites, or moving the other nodes, they move to different locations other than follow the attached node?

**Comment by pigne** [comment]
Could you give a minimal example that reproduces this behavior and post it on a gist or something?

**Comment by PlayMa256** [comment]
Sure,  [check it here](https://gist.github.com/PlayMa256/3e7161484334d7becee1d56fc2c51d22)

**Comment by pigne** [comment]
Thank for spotting that error and helping with the example. 

I push the correction on the master branch. 

You can also wait for the nightly build binaries within 24 hours at https://data.graphstream-project.org/pub/1.x/nightly-build/last/

**Comment by PlayMa256** [comment]
Thank you and its been more than 24hrs and the binaries still not there.

**Comment by pigne** [comment]
Our automatic build system is down for now. I am afraid you will have to build the library. 

I suggest you checkout branch `1.x` and then build it with maven: http://graphstream-project.org/doc/Tutorials/GraphStream-Maven/#building-graphstream-using-maven

**Comment by PlayMa256** [comment]
i dont know if i should keep this discussion here, but still related with the compile of the jar file. I was using the latest 1.3 version and my code was running ok.
 
I was able to compile the gs-core, followed all the steps and compiled using the nodeps profile which according to the website, packs some dependencies with it. After it, a version 2.0 was created successfully, it seemed little weird because the latest version still 1.3 but i moved on. 

Replaced on my project, imported it correctly and now it's saying a class was not found, the class is: `Exception in thread "AWT-EventQueue-0" java.lang.NoClassDefFoundError: org/graphstream/ui/swingViewer/GraphRenderer`. I went to both 1.3 and 2.0 jars to check what was inside, and apparently there is a difference between them, [you can check it on this image.](http://image.prntscr.com/image/990bc495b6714acd946ff231fbbd1163.png)

Now i might ask, was this class removed from the latest versions or did my compilation process go wrong?

**Comment by pigne** [comment]
OK you are probably using `gs-ui` which needs to be updated too. 

You don't want any 2.0-related build yet, so stick with branch `1.x`:

1. checkout branches  `1.x`  on both gs-core and gs-ui
(`git checkout 1.x`)
2. recompile `gs-core` with maven as you already did and install it locally (`mvn install`) in order to get `gs-core-1.3.1-SNAPSHOT.jar`
3. recompile `gs-ui` the same way in order to get `gs-ui-1.3.1-SNAPSHOT.jar`

**Comment by PlayMa256** [comment]
I did everything as you said and now i've got a different error. Im answering later because i did a few tests. I was getting some errors and apparently it was because i had java8 as default on my system and it wasn't able to find few dependencies.

Switched up to `java-7-openjdk-amd64` and compiled correctly with a few deprecation warnings. If necessary i can post it later.

Ok, did exactly on the order you said.
- downloaded gs-deps, `mvn install`
- cloned gs-core `1.x` and compiled correctly with `mvn install` to install locally and with the profile nodeps to production.
- cloned gs-ui `1.x`, `mvn install` and generated the jar file.

Ok, moved all the new dependencies to my project, no error so far, but when i launch the visualization a new error appears: `java.lang.NoClassDefFoundError: scala/collection/convert/WrapAsScala`

I know i should not have scala installed on my computer because maven already uses a scala compiler to make our lifes easier and i have no idea what could be the problem. I appreciate the help and patience. Thank you.

**Comment by pigne** [comment]
Do you use maven to compile your own projet too? From what you said (*"[...] moved all the new dependencies to my project [...]"*) I guess you don't. 

The easiest way for you now that you already use maven is to use it in your project too. 

If you don't want to use maven then you need to add all the dependencies of gs-ui and gs-core in your CLASSPATH by yourself. 

If you'r OK with maven, then create a `pom.xml` (many examples are available online) at the root of your project and specify the following dependencies : 
```xml
<dependencies>
	<dependency>
		<groupId>org.graphstream</groupId>
		<artifactId>gs-core</artifactId>
		<version>1.3.1-SNAPSHOT</version>
		<optional>false</optional>
	</dependency>
	<dependency>
		<groupId>org.graphstream</groupId>
		<artifactId>gs-ui</artifactId>
		<version>1.3.1-SNAPSHOT</version>
		<optional>false</optional>
	</dependency>
</dependencies> 
```

Note that since you compiles `gs-core` and `gs-ui` with `mvn install` the dependencies will be automatically available for your own project. You don't need to move the generated jars you your project. Maven will handle it for you.

**Comment by PlayMa256** [comment]
I have inherit based on a research/development i'm doing for a project at my University and it started as a student's project and went through the time without maven thats why i don't use it, i might start using because of graphstream and it is going to help future improvements too.

I'm not that into maven so, explain me something. Assume i'm running this on windows (thats because 99% of the future students is going to run on windows too), i will compile gs-core and gs-ui so it will be available locally for me, when the other students are going to run the same project, will they have to compile gs-core and ui again in order to have the newest version? Assuming they will use it before your system come back.  (i know that when your build system come back the newest version of gs-xx will be available through maven).

**Comment by pigne** [comment]
The `mvn install` command only populates your local maven repository on your own machine, not on any other machine.

As I said earlier if maven is a no-go then prefer the copy/paste solution. Distribute the newly compiled jar files of `gs-core` and `gs-ui` as well as their dependencies so your students.

- `gs-core` will need `phred` and `mbox` (you compiled them in the `gs-deps` package) 
- `gs-ui` needs `gs-core` obviously, `gs-algo` (keep the official release) and `scala-library` version 2.10.1

FYI running `mvn dependency:tree` at the root of a maven projet will give you its full dependency tree.

**Comment by PlayMa256** [comment]
I followed your advice and installed with maven and with solo dependencies + scaba-library. Both worked but the bug still happening, the sprites were moving in the X axis, now they move freely. You can check a test [here](https://youtu.be/-QZ2gw4HJjU).

**Comment by pigne** [comment]
Well I beg to differ on that comment.

- Sprites can be moved with the mouse.
- But their position is also relative to their attached node. Move the node and the sprite will move accordingly/relatively.
 
If you click on the sprite, you'll move the sprite. If you want to move both the sprite and the node then click on the node. 

I don't know about your precise scenario or your precise needs but from a general point of view (keep in ming that GraphStream is a general-purpose dynamic graph library) I believe this is the desired behavior.

**Comment by PlayMa256** [comment]
Well, i understand and appreciate the real work and effort put into graphstream, i'm just discussing some features and try to help as much as i can, even seeing i'm just complaining.

I've checked again and looked into a few old issues and found the #80 which talks about imprecision whilst clicking on nodes, so i went to check again my [visualization](https://youtu.be/eChSksy0u7c), clicking correctly on the node and i was able just to move the sprite. Was the #80 discussed or fixed? 

I used some styles on my sprites increasing the size/padding of it, i dont know if it makes any difference at all but, its worth asking.
